pkg/kubectl/cmd/util/helpers.go

```diff
@@ -113,48 +117,46 @@ func checkErrWithPrefix(prefix string, err error) {
 	checkErr(prefix, err, fatalErrHandler)
 }
 
-func checkErr(pref string, err error, handleErr func(string)) {
-	if err == nil {
+func checkErr(pref string, err error, handleErr func(string, int)) {
+	switch {
+	case err == nil:
 		return
-	}
-
-	if errors.IsInvalid(err) {
+	case errors.IsInvalid(err):
 		details := err.(*errors.StatusError).Status().Details
-		prefix := fmt.Sprintf("%sThe %s %q is invalid.\n", pref, details.Kind, details.Name)
+		prefix := fmt.Sprintf("%sThe %s %q is invalid", pref, details.Kind, details.Name)
 		errs := statusCausesToAggrError(details.Causes)
-		handleErr(MultilineError(prefix, errs))
-	}
-
-	if noMatch, ok := err.(*meta.NoResourceMatchError); ok {
-		switch {
-		case len(noMatch.PartialResource.Group) > 0 && len(noMatch.PartialResource.Version) > 0:
-			handleErr(fmt.Sprintf("%sthe server doesn't have a resource type %q in group %q and version %q", pref, noMatch.PartialResource.Resource, noMatch.PartialResource.Group, noMatch.PartialResource.Version))
-		case len(noMatch.PartialResource.Group) > 0:
-			handleErr(fmt.Sprintf("%sthe server doesn't have a resource type %q in group %q", pref, noMatch.PartialResource.Resource, noMatch.PartialResource.Group))
-		case len(noMatch.PartialResource.Version) > 0:
-			handleErr(fmt.Sprintf("%sthe server doesn't have a resource type %q in version %q", pref, noMatch.PartialResource.Resource, noMatch.PartialResource.Version))
+		handleErr(MultilineError(prefix, errs), DefaultErrorExitCode)
+	case clientcmd.IsConfigurationInvalid(err):
```

Comment:
If ``clientcmd.IsConfigurationInvalid(err)`` and ``noMatch, ok := err.(*meta.NoResourceMatchError); ok`` can hold at the same time, order of testing should not be switched.